### DISCUSSION TITLE
Unable to upload artifact - OpenSslLamdaLayer resource

### DISCUSSION THREAD

**Discussion by biplab123321** [aws/aws-connected-device-framework#91] [General]
using installer to loadmodules but gets the below error msg:

Unable to upload artifact ../build/build.zip referenced by ContentUri parameter of OpenSslLamdaLayer resource
Attaching the terminal commands and details.
[error.txt](https://github.com/aws/aws-connected-device-framework/files/9913426/error.txt)
(url)
Have run all the step of rush install and rush build steps.Request to hep or let me know if i missed any step.

**Comment by aaronatbissell** [comment]
Have you done a `rush bundle`? It appears that your source for the OpenSslLambdaLayer isn't bundled into the .zip file required to be uploaded to the lambda. This happens during the `rush bundle` step in the process.

https://github.com/aws/aws-connected-device-framework/blob/main/source/docs/deployment.md#deploying-from-source

**Reply by biplab123321** [reply]
Rush bundlecomand have he below 
error message:
osboxes@osboxes:~/aws-connected-device-framework/source$ rush bundle


Rush Multi-Project Build Tool 5.47.0 - https://rushjs.io
Node.js version is 16.13.0 (LTS)


Starting "rush bundle"


**********************************************************
  Building the OpenSSL lambda layer
**********************************************************

Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.39/build?buildargs=%7B%7D&cachefrom=%5B%5D&cgroupparent=&cpuperiod=0&cpuquota=0&cpusetcpus=&cpusetmems=&cpushares=0&dockerfile=Dockerfile&labels=%7B%7D&memory=0&memswap=0&networkmode=default&rm=1&session=p9lj005529n5lvfcmxfn3ba8w&shmsize=0&t=cdf-openssl-lambda-layer&target=&ulimits=null&version=1: dial unix /var/run/docker.sock: connect: permission denied

The script failed with exit code 1
osboxes@osboxes:~/aws-connected-device-framework/source$

any help on how to resolve this.

**Reply by biplab123321** [reply]
https://stackoverflow.com/questions/48957195/how-to-fix-docker-got-permission-denied-issue
I followed this link and able to resolve is. Thank you :)

**Reply by biplab123321** [reply]
The deployment was successful. I was testing the rest call from API Gateway to do Greengrass provisioning.
The API generated for the greengrass provisioning: has the error : Method completed with status: 415

attaching error let me know if I am missing something in the POST Call.
![error](https://user-images.githubusercontent.com/114371445/199797406-0f07fcc0-ce52-4566-a0f8-b1e2a28d4ed2.png)

Request to let me know if any thing i am missing. the same was faced with postman too.
